framework/src/main/java/org/apache/drill/test/framework/TestDriver.java

```diff
@@ -182,7 +188,7 @@ public int runTests() throws Exception {
       if (cmdParam.trackMemory) {
     	  queryMemoryUsage();
       }
-
+      failedTests.clear();
```

Comment:
Rather than setting and clearing in each iteration, if any test fails in any iteration, we should capture all those tests in a HashSet or something similar, and at the end of all iterations, we should rerun these to isolate random failures.